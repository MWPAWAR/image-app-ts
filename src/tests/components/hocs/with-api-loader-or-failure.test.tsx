import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import withApiLoaderOrFailure from '../../../components/hocs/with-api-loader-or-failure'

describe('withApiLoaderOrFailure', () => {
  const mockStore = configureStore([])

  it('Matches snapshot with loading state', () => {
    const store = mockStore({
      images: {
        images: [],
        status: 'loading',
      },
    })

    const Component = () => <div>Blah</div>
    const MockComponent = withApiLoaderOrFailure(Component)

    const tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <MockComponent activeTab="recentlyAdded" />
        </Provider>
      </MemoryRouter>
    )
    expect(tree).toMatchSnapshot()
  })

  it('Matches snapshot with error state', () => {
    const store = mockStore({
      images: {
        images: [],
        status: 'error',
      },
    })

    const Component = () => <div>Blah</div>
    const MockComponent = withApiLoaderOrFailure(Component)

    const tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <MockComponent activeTab="recentlyAdded" />
        </Provider>
      </MemoryRouter>
    )
    expect(tree).toMatchSnapshot()
  })

  it('Matches snapshot with success state', () => {
    const store = mockStore({
      images: {
        images: [],
        status: 'success',
      },
    })

    const Component = () => <div>Blah</div>
    const MockComponent = withApiLoaderOrFailure(Component)

    const tree = renderer.create(
      <MemoryRouter>
        <Provider store={store}>
          <MockComponent activeTab="recentlyAdded" />
        </Provider>
      </MemoryRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})
