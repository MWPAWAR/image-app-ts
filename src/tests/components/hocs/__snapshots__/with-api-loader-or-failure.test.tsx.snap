// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withApiLoaderOrFailure Matches snapshot with error state 1`] = `
<div
  className="loaderOrErrorContainer"
>
  Something went wrong
</div>
`;

exports[`withApiLoaderOrFailure Matches snapshot with loading state 1`] = `
<div
  className="loaderOrErrorContainer"
>
  Loading...
</div>
`;

exports[`withApiLoaderOrFailure Matches snapshot with success state 1`] = `
<div>
  Blah
</div>
`;
